File: leetcode/reverseList/1.js
function ListNode(val) { 
    this.val = val 
    this.next = null 
}
var reverseList = function (head){
    if(!head || !head.next) return head;
    let cur = head
    let pre = null
    while(cur){
        let next = cur.next
        cur.next  = pre
        pre = cur
        cur = next
    }   
    return pre
}

const n1 = new ListNode(1) 
const n2 = new ListNode(2)
const n3 = new ListNode(3)
const n4 = new ListNode(4)
const n5 = new ListNode(5)
n1.next = n2;
n2.next = n3;
n3.next = n4;
n4.next = n5;


console.log(reverseList(n1))

File: leetcode/reverseList/1.py
class ListNode: 
    def __init__(self,val):
        self.val = val
        self.next = None 
    def __repr__(self): # 覆盖类里面的方法
        if self:
            return "{}-{}".format(self.val,repr(self.next))

class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        if not head: return None
        prev = None
        cur = head
        while cur:
            cur.next,prev,cur = prev,cur,cur.next
        return prev

n1 = ListNode(1)
n2 = ListNode(2)
n3 = ListNode(3)
n4 = ListNode(4)
n5 = ListNode(5)

n1.next = n2
n2.next = n3
n3.next = n4
n4.next = n5

x = Solution()
print(x.reverseList(n1))